### PR TITLE
Dac validity param

### PR DESCRIPTION
**Pull request by lrrash**
I added dacValidityInDays as a parameter into CDK/CFN and made it an env var in the Lambda.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,13 +93,17 @@ When deploying the application, you can configure it to support different use ca
    $ echo "One way would be to poll s3 key for presence of .pem or .err (should a failure happen) files" 
 
    $ aws s3 cp s3://matterstackpai-dacinputs3tosqss3bucket<remainder of your bucket name>/arn:aws:acm-pca:<region>:<account>:certificate-authority/<PAI UUID>/<PID>/cert.pem .
+
+   $ echo "Change the DAC validity as needed"
+   $ aws lambda update-function-configuration --function-name <DAC_ISSUING_LAMBDA_NAME> --environment 'Variables {dacValidityInDays=<VALIDITY_IN_DAYS>}'
```
Comment (hmphome) [inline README.md:98]: Have you checked that it's safe to remove whatever variables are being added by default (as this command removes them all)?

File: lib/matterStacks.ts [hunk 2]
```diff
@@ -99,6 +99,10 @@ export class MatterStack extends Stack {
                 type: "Number",
                 description: "Validity in days for new PAI(s)"
             }).valueAsNumber;
+            const dacValidityInDays = new CfnParameter(this, "dacValidityInDays", {
```
Comment (hmphome) [inline lib/matterStacks.ts:102]: Let's use https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.CfnRuleAssertion.html somewhere below to check that the two validity intervals make sense.

File: lib/matterStacks.ts [hunk 3]
```diff
@@ -99,6 +99,10 @@ export class MatterStack extends Stack {
...
+                type: "String",
```
Comment (hmphome) [inline lib/matterStacks.ts:103]: Not a string, a number! 🙂

File: README.md [hunk 4]
```diff
@@ -93,13 +93,17 @@ When deploying the application, you can configure it to support different use ca
...
    ```
 
 ### Parameters
 1. `--parameters vendorId=<VID>` - The vendor ID to be assigned to the CA. Note that when creating PAIs this value should be the same as in the PAA.
```
Comment (hmphome) [inline README.md:102]: while we're already here, let's mention that it's a 4-digit hex value

File: lib/matterStacks.ts
Comment (hmphome) [inline lib/matterStacks.ts:1]: Let's mention in "vendorId" and "productIds" parameters that those are 4-digit hex values..

**Review by hmphome — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (lrrash) [reply README.md:98]: I don't think there are any variables being added be default. This is the only env var when i first create the lambda. But also yes I did run the cli command to update the env var and it still worked

File: lib/matterStacks.ts (hunk 2 above)
Comment (lrrash) [reply lib/matterStacks.ts:102]: Im not sure if this is possible. The only cfn condition we can check is Equals

**Review by hmphome — CHANGES_REQUESTED** [review]

File: lib/matterStacks.ts [hunk 5]
```diff
@@ -89,16 +89,20 @@ export class MatterStack extends Stack {
             // Create PAI
             let prodIds = new CfnParameter(this, "productIds", {
                 type: "List<String>",
-                description: "A list of product IDs to create first PAIs"
+                description: "A list of product IDs to create first PAIs. These must be 4-digit hex values."
             })?.valueAsList;
             const vendorId = new CfnParameter(this, "vendorId", {
                 type: "String",
-                description: "If paaArn parameter is not provided, then this vendorId is used to create PAI(s)"
+                description: "If paaArn parameter is not provided, then this vendorId is used to create PAI(s). This must be a 4-digit hex value."
```
Comment (hmphome) [inline lib/matterStacks.ts:96]: Interesting, paaArn actually cannot be not provided when creating PAIs..

Sorry, haven't noticed it before.

**Review by hmphome — CHANGES_REQUESTED** [review]

**Pull request merged**